packages/types/src/products.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/**
+ * - `null` - explicitly Channels
+ * - `string` - uuid - any other product
+ */
+export type ProductIdentifier = null | string;
```

Comment:
I started down a path of converting this to a meaningful id, but ended up reverting it to minimize changes as this was already an established pattern. 

Though I could see this might be a good time to change it if we were going to. Let's bring it up in web guild perhaps and get some thoughts.